### DISCUSSION TITLE
Summary

### DISCUSSION THREAD

**Discussion by bameyrick** [ONSdigital/design-system#1386] [Feedback]
Use this place to discuss the [summary component](https://ons-design-system.netlify.com/components/summary) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by bameyrick** [comment]
The breakpoint for changing from layout to have question, answer, and action all on one row, vs question above answer and action is currently set to > 25 chars for the answer length. This could probably be bumped up to 50 chars.

**Comment by bameyrick** [comment]
As discussed with @will-madebymoon, the answer column needs a greater width bias, but still be flexible for translations/different actions etc.

**Comment by bameyrick** [comment]
As discussed with @will-madebymoon, we should remove the mode for wrapping answers below questions on desktop.

**Comment by will-madebymoon** [comment]
Move the answer over in the summary providing more space to the question and so that the change link is closer to the actual answer
<img width="730" alt="Screenshot 2019-04-02 at 09 34 40" src="https://user-images.githubusercontent.com/8184622/55388150-a7941d00-552a-11e9-999a-8cd4d0080918.png">

**Comment by bameyrick** [comment]
The condition that checks if a rowItem has values does not work in jinja2

**Comment by bameyrick** [comment]
The action spacer element (pipe) is being read by screen readers.

**Comment by bameyrick** [comment]
When the summary is within a question (e.g. household collector), the first row should not have a top border and the last item should not have a bottom border.

**Comment by bameyrick** [comment]
As discussed with @will-madebymoon and @boxadesign we should remove the top border from the first row, and the bottom border from the last row.

**Comment by bameyrick** [comment]
We need to add the top border back on to the first item always so that zoom text users can discover the action links.

**Comment by jrbarnes9** [comment]
The HTML used to emphasise parts of a question title has been included in the aria-label associated with the ‘Change’ link on the summary. This could confuse some users navigating out of
context who may be unaware of the purpose or meaning of the HTML. 
```html
<td class="summary__actions">
<a href="/questionnaire/accommodation-type/#accommodation-type-answer"
class="summary__button" aria-label="Change your answer for: What type of
accommodation is <em>68 Abingdon Road, Goathill</em>?" data-qa="accommodationtype-answer-edit">Change</a>
</td>
```

**Comment by jrbarnes9** [comment]
Inconsistent borders in household collector summaries...
<img width="712" alt="Screenshot 2020-01-22 at 16 42 49" src="https://user-images.githubusercontent.com/43346934/72914519-b3a9cf00-3d36-11ea-9dd4-ac5c76d285da.png">
...proposing to always display a bottom border on a summary if a link follows...
<img width="742" alt="Screenshot 2020-01-22 at 16 47 21" src="https://user-images.githubusercontent.com/43346934/72914681-f2d82000-3d36-11ea-871f-c9328fed66fe.png">
<img width="717" alt="Screenshot 2020-01-22 at 16 47 37" src="https://user-images.githubusercontent.com/43346934/72914682-f2d82000-3d36-11ea-99f3-245d3460b9e7.png">

**Comment by JNATION85** [comment]
Are we able to increase the green tick on summary page by 1px? I think it needs to be more prominent in the hub 

<img width="877" alt="Screenshot 2020-02-13 at 11 36 15" src="https://user-images.githubusercontent.com/41283876/74430616-1d5c5b00-4e55-11ea-9d7c-edfd8a1c68fa.png">

**Comment by jrbarnes9** [comment]
Multiple/grouped summary example is broken

**Comment by jrbarnes9** [comment]
Long words breaking summary

<img width="622" alt="Screenshot 2020-06-08 at 08 58 55" src="https://user-images.githubusercontent.com/43346934/84006290-64231300-a966-11ea-9639-e9ffc37c4fae.png">

**Comment by jrbarnes9** [comment]
All summary values should not be bulleted
<img width="971" alt="Screenshot 2020-08-03 at 11 28 39" src="https://user-images.githubusercontent.com/43346934/89173809-b9e9f500-d57c-11ea-8887-e05968f09f57.png">

**Comment by jrbarnes9** [comment]
Reduce top padding on the link container that follows a summary 

Should be `u-pt-s` not `u-pt-m`

From this...
![Screenshot 2020-10-27 at 08 28 58](https://user-images.githubusercontent.com/43346934/97276176-c48bf180-182e-11eb-8fea-4ade08d85093.png)

To this...
![Screenshot 2020-10-27 at 08 29 11](https://user-images.githubusercontent.com/43346934/97276203-cb1a6900-182e-11eb-95b0-89e19a6b6c2a.png)

This relates to the work done to fix this issue...
> Inconsistent borders in household collector summaries...
> <img alt="Screenshot 2020-01-22 at 16 42 49" width="712" src="https://user-images.githubusercontent.com/43346934/72914519-b3a9cf00-3d36-11ea-9dd4-ac5c76d285da.png">
> ...proposing to always display a bottom border on a summary if a link follows...
> <img alt="Screenshot 2020-01-22 at 16 47 21" width="742" src="https://user-images.githubusercontent.com/43346934/72914681-f2d82000-3d36-11ea-871f-c9328fed66fe.png">
> <img alt="Screenshot 2020-01-22 at 16 47 37" width="717" src="https://user-images.githubusercontent.com/43346934/72914682-f2d82000-3d36-11ea-99f3-245d3460b9e7.png">

**Comment by Al-Davies** [comment]
#Summary User Research
tested as part of eQ usability testing 7 March 2022 (testing involved creation of hub page with sections asking participants to provide monetary values spend on certain goods/services)

* Participants were clear on reaching the initial summary page that they could review and change their answers
  * Changes to answers resulted in updating of the overall spend on a group of items e.g. food, participants understood this was happening
* Once initial four sections had been completed, another section was added to the hub, which summarised the previous sections where monetary values had been added.
  * Participants didn’t naturally question this, those who just viewed it understood it’s purpose. Those who were asked their expectations around this newly added section thought it may contain more questions, but again, understood it’s purpose when they went into the section.
* Participants saw the overall summary as a triple check of their answers
* Again, participants all able to change previously entered values from the overall summary screen, moving through utility bills section to amend total given for water.

**Comment by jrbarnes9** [comment]
Summary parameter `rowTitleAttributes` should be part of the `row` object, not the `rowItem` object

So it sits alongside the `rowTitle`.

**Reply by jrbarnes9** [reply]
#2275 #2185

**Comment by jrbarnes9** [comment]
`.ons-summary__spacer` does not match the height of the action links
![Screenshot 2022-10-24 at 10 00 40](https://user-images.githubusercontent.com/43346934/197489127-4c868f1e-99a0-45f7-bef8-40fbc4b8f81e.png)

**Reply by jrbarnes9** [reply]
Fixed by #2506

**Comment by jrbarnes9** [comment]
eQ recently encountered new use cases with list collectors within grouped summaries. It would be worth reviewing the recent enhancement to the gov.uk design system as a possible solution: https://github.com/alphagov/govuk-design-system-backlog/issues/210
